### PR TITLE
generate_codon_frq_CFmethod

### PR DESCRIPTION
**Pull request by evfaust**
- defined the method generate_codon_frq in ComputeFeatures class to populate class variables codon_counts and codon_frqs with dictionaries representing codon bias of each virus and host
- imported gene_features functionality for incorporation
- defined pytest for generate_codon_frq
- after approval, I'll model this format when defining functions for RSCU, aa_freq, etc, and the populated class variables from these methods will then be compared between virus-host Pairs in the compute_features method

### CODE REVIEW

File: src/vhip/mlmodel/compute_ml_features.py [hunk 1]
```diff
@@ -300,6 +301,42 @@ def generate_kmer_profiles(self):
             seq_profile.generate_profile()
             self.k6profiles[host] = seq_profile
 
+    def generate_codon_frq(
+        self, threshold_imprecise: float = 0.0, threshold_skipped_genes: float = 0.5
+    ) -> None:
+        """Generate profile of the frequency of each unique codon in every virus and host GeneSet.
+
+        This will be done for each .ffn files in the virus and host genes files directories.
```
Comment (GBastion) [inline src/vhip/mlmodel/compute_ml_features.py:309]: Can you clarify this? Did you mean for each genes file in the .ffn file?

File: src/vhip/mlmodel/compute_ml_features.py [hunk 2]
```diff
@@ -300,6 +301,42 @@ def generate_kmer_profiles(self):
...
+
+        Args:
+            threshold_imprecise (float): Percentage of imprecise (non-ATGC) codons tolerated in a single gene (default 0.0 or 0%)
```
Comment (GBastion) [inline src/vhip/mlmodel/compute_ml_features.py:312]: can you make a note on how you determined those values?

File: tests/test_ComputeFeatures.py
Comment (GBastion) [inline tests/test_ComputeFeatures.py:1]: please add a test to check the codon frequency was computed correctly! 

like if the expected value for a codon is 0.3 then make sure it is that number

**Review by GBastion — CHANGES_REQUESTED** [review]

File: src/vhip/mlmodel/compute_ml_features.py (hunk 1 above)
Comment (evfaust) [reply src/vhip/mlmodel/compute_ml_features.py:309]: Changed. Does that make more sense?

File: src/vhip/mlmodel/compute_ml_features.py (hunk 2 above)
Comment (evfaust) [reply src/vhip/mlmodel/compute_ml_features.py:312]: see methods. let me know if the decision we made still makes sense!

File: tests/test_ComputeFeatures.py
Comment (evfaust) [reply tests/test_ComputeFeatures.py:1]: added to the existing test for generate_codon_frq

**Review by GBastion — APPROVED** [review]

**Pull request merged**